### ISSUE TITLE
feat(timepicker) - add keyboard navigation

### ISSUE THREAD

**Issue by nadavten** [valor-software/ngx-bootstrap#1007]
# bug

timer - arrowKeys not working.

and i dont see the arrows (up and down) for each input of hour:minutes

**Comment by zacharybil** [comment]
Setting [arrowkeys]="false" doesn't seem to change any behavior for me either.

Are you sure you don't have [showSpinners]="false"? That removes the up down arrow chevrons.

**Comment by nadavten** [comment]
Im sure

**Comment by SystemDisc** [comment]
Pressing the arrow keys on my Macbook Air (macOS Sierra) in the latest Chrome does not change the value in the timepicker, even if I explicitly state `[arrowkeys]="true"`. It does not work on the demo page for me, either: https://valor-software.com/ng2-bootstrap/#/timepicker

**Comment by SystemDisc** [comment]
It may be that pressing up by default puts the cursor at the beginning of the input and down puts it at the end. Perhaps you need to `e.preventDefault();`

**Comment by rishuvaid** [comment]
'arrowkeys' property is not working for me either. Is it fixed yet? or is there something I may be missing?

**Comment by SystemDisc** [comment]
It's still not working.

I ended up putting a hacky jQuery script in a plain JS file that I link to in my header.

```javascript
$(function() {
	$(document).on('keydown', 'timepicker input[type=text], timepicker button', function(e) {
		if (e.keyCode === 38 || e.keyCode === 40) {
			e.preventDefault();
			e.stopPropagation();
			let $this = $(e.currentTarget);
			let index = $this.closest('td').index();
			if ($this.is(':not(button)')) {
				if (e.keyCode === 38) {
					$this.closest('table').find('tr').first().find('td:nth-child(' + (index + 1) + ') a > *').first().click();
				}
				else {
					$this.closest('table').find('tr').last().find('td:nth-child(' + (index + 1) + ') a > *').first().click();
				}
			}
			else {
				$this.click();
			}
		}
	});
});
```

This works even with the buttons hidden.

**Comment by IlyaSurmay** [comment]
fixed in #2058
